rust/parser/src/lexer.rs

```diff
@@ -1607,6 +1612,10 @@ impl<'alloc> AutoCow<'alloc> {
         ));
     }
 
+    fn has_different(&self) -> bool {
```

Comment:
This could use a comment:

`check if the string contains a different character, such as an escape sequence`